### ISSUE TITLE
[Patch]: Flaky test on OSX Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.AddressRegistrationTests.RegisterAddresses_IPv4_Success

### ISSUE THREAD

**Issue by pranavkm** [aspnet/KestrelHttpServer#1729]
```C#
System.Net.Sockets.SocketException : Network is unreachable
   at Microsoft.AspNetCore.Testing.HttpClientSlim.<GetSocket>d__7.MoveNext() in /Users/asplab/TeamCity/work/d6337ac2abf63a4b/.repositories/KestrelHttpServer/test/shared/HttpClientSlim.cs:line 114
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Testing.HttpClientSlim.<GetStream>d__6.MoveNext() in /Users/asplab/TeamCity/work/d6337ac2abf63a4b/.repositories/KestrelHttpServer/test/shared/HttpClientSlim.cs:line 85
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Testing.HttpClientSlim.<GetStringAsync>d__1.MoveNext() in /Users/asplab/TeamCity/work/d6337ac2abf63a4b/.repositories/KestrelHttpServer/test/shared/HttpClientSlim.cs:line 24
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.AddressRegistrationTests.<RegisterAddresses_Success>d__7.MoveNext() in /Users/asplab/TeamCity/work/d6337ac2abf63a4b/.repositories/KestrelHttpServer/test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/AddressRegistrationTests.cs:line 93
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
```

**Comment by pranavkm** [comment]
cc @Eilon \ @muratg

**Comment by natemcmaster** [comment]
@pranavkm how long has this been flaky? Trying to determine if this was before/after #1717

**Comment by natemcmaster** [comment]
Wait, just read the title again. Which patch build is this for? 1.0.x or 1.1.x?

**Comment by pranavkm** [comment]
This was the 1.1.x patch. But if there are code changes to stablilize the test, it'd be great to get them in both the patch branches.

**Comment by pranavkm** [comment]
Let me know when you've had a chance to skip \ fix this test. I'll redo a patch build.

**Comment by halter73** [comment]
"Network is unreachable." This isn't even saying the host is unreachable, it's saying the *network* is unreachable. Wouldn't almost any functional test fail if this is the case? I don't think it makes sens to skip just AddressRegistrationTests for this kind of failure.

**Comment by natemcmaster** [comment]
The flakiness is likely caused by this test data:
https://github.com/aspnet/KestrelHttpServer/blob/rel/1.1.2/test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/AddressRegistrationTests.cs#L229
```c#
dataset.Add($"http://{Dns.GetHostName()}:0/;https://{Dns.GetHostName()}:0/", GetTestUrls);
```
Everything else in the test uses loopback.

**Comment by natemcmaster** [comment]
Resolved in #1767 and #1762
